pkg/controller/registry/resolver/step_resolver.go

```diff
@@ -193,14 +194,56 @@ func (r *OperatorStepResolver) ResolveSteps(namespace string) ([]*v1alpha1.Step,
 			})
 		}
 
-		// add steps for subscriptions for bundles that were added through resolution
+		// add steps for subscriptions for bundles that were added through resolution and bundles that have been marked deprecated
 		for sub := range existingSubscriptions {
-			if sub.Status.CurrentCSV == op.Name {
-				continue
+			updated := false
+			if len(op.Deprecations) == 0 && sub.Status.GetCondition(v1alpha1.SubscriptionDeprecated).Status != corev1.ConditionUnknown {
+				// If new bundle is no longer deprecated then remove the conditions
+				sub.Status.RemoveConditions(v1alpha1.SubscriptionDeprecated, v1alpha1.SubscriptionPackageDeprecated, v1alpha1.SubscriptionChannelDeprecated, v1alpha1.SubscriptionBundleDeprecated)
+				updated = true
+			} else if len(op.Deprecations) > 0 {
+				// Remove old conditions first in case of update
+				sub.Status.RemoveConditions(v1alpha1.SubscriptionDeprecated, v1alpha1.SubscriptionPackageDeprecated, v1alpha1.SubscriptionChannelDeprecated, v1alpha1.SubscriptionBundleDeprecated)
+				rollupMessage := ""
+				for _, deprecation := range op.Deprecations {
+					switch deprecation.Type {
+					case declcfg.SchemaPackage:
+						sub.Status.SetCondition(v1alpha1.SubscriptionCondition{
+							Type:    v1alpha1.SubscriptionPackageDeprecated,
+							Message: deprecation.Message,
+							Status:  corev1.ConditionTrue,
+						})
+					case declcfg.SchemaChannel:
+						sub.Status.SetCondition(v1alpha1.SubscriptionCondition{
+							Type:    v1alpha1.SubscriptionChannelDeprecated,
+							Message: deprecation.Message,
+							Status:  corev1.ConditionTrue,
+						})
+					case declcfg.SchemaBundle:
+						sub.Status.SetCondition(v1alpha1.SubscriptionCondition{
+							Type:    v1alpha1.SubscriptionBundleDeprecated,
+							Message: deprecation.Message,
+							Status:  corev1.ConditionTrue,
+						})
+					}
+					rollupMessage = fmt.Sprintf("%s; %s", rollupMessage, deprecation.Message)
+				}
+				sub.Status.SetCondition(v1alpha1.SubscriptionCondition{
+					Type:    v1alpha1.SubscriptionDeprecated,
+					Message: rollupMessage,
+					Status:  corev1.ConditionTrue,
+				})
+				updated = true
```

Comment:
If the sub already has this existing set of deprecations and none of the messages changed, it seems like we should not set `updated = true`?

Reply:
Yes, I'm updating this to detect if a change occurs and only update when required.